lib/game_turn.rb

```diff
@@ -0,0 +1,34 @@
+module TicTacToe
+  class GameTurn
+
+    def initialize(board, console_ui, eval_game, validation)
+      @board = board 
+      @console_ui = console_ui 
+      @eval_game = eval_game
+      @validation = validation
+    end
+
+    def execute(player, marker)
+      move = @console_ui.move_messages(@board.nine_space_array, player, marker)
+      move = valid_move_loop(move)
+      @board.fill(move, marker)
+      if @eval_game.won?(@board.nine_space_array)
+        @console_ui.display_board(@board.nine_space_array)
+        player
+      elsif @eval_game.tied?(@board.nine_space_array)
+        @console_ui.display_board(@board.nine_space_array)
+        0
+      end
+    end
+    
+    private
+
+    def valid_move_loop(move)
+      until @validation.board_conditions?(@board.nine_space_array, move)
```

Comment:
Can you think of a better name for the `@validation.board_conditions?` method? Without looking at it's implementation, it's not completely clear as to what the method is actually doing by the name